File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#include <deque>
#include <unordered_map>
using namespace std;

string solution(string s, string skip, int index) {
    sort(skip.begin(), skip.end());
    deque<char> de(skip.begin(), skip.end());    
    
    unordered_map<char, int> alpha1;
    vector<char> alpha2;
    
    int cnt = 0;
    for (int i = 0; i < 26; i++){
        char c = i + 'a';
        
        if (!de.empty() && c == de.front()){
            de.pop_front();
            continue;
        }
        
        alpha1[i + 'a'] = cnt++;
        alpha2.push_back(i + 'a');
    }

    string answer = "";
    
    for (char c : s){
        int idx = (alpha1[c] + index) % cnt;
        answer += alpha2[idx];
    }

    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    
    alpa = [chr(c) for c in range(ord('a'), ord('z') + 1) if chr(c) not in skip]
    
    ans = ''
    
    for c in s:
        idx = (alpa.index(c) + index) % len(alpa)
        ans += alpa[idx] 

    
    return ans